src/documentation/layout.pug

```diff
@@ -14,3 +15,43 @@ html
     :reload
     script(src='[[TIKUI_BASEPATH]]lib/prismjs/prism.js')
     script(src='[[TIKUI_BASEPATH]]lib/prismjs/prism-pug.js')
+    script.
+      document.addEventListener('DOMContentLoaded', initNav, false);
+
+      function initNav () {
+        const content = document.querySelector('.tikui-doc-content');
+        const nav = document.querySelector('.tikui-doc-nav');
+
+        const titles = content?.querySelectorAll('h1, h2, h3, h4, h5, h6') || [];
+
+        // build nav
+        titles.forEach((title, index) => {
+          title.setAttribute('data-tikui-doc-nav-index', index);
+          const level = title.tagName.slice(1);
+          nav.innerHTML += `
+            <div>
+              <a class="tikui-doc-nav-item -level-${level}" data-tikui-doc-nav-scrollTo="${index}">${title.innerText}</a>
+            </div>
+          `;
+        });
+
+        // scroll to target element on click
+        document.querySelectorAll('[data-tikui-doc-nav-scrollTo]').forEach(el => {
+          el.addEventListener('click', e => {
+            e.preventDefault();
+
+            const targetIndex = el.getAttribute('data-tikui-doc-nav-scrollTo');
+            const target = document.querySelector(`[data-tikui-doc-nav-index="${targetIndex}"]`);
+
+            if (target) {
+              const headerOffset = 56 + 15;
```

Comment:
It could be interesting to use anchors instead of query selector because in the Tikui style, when you go to an anchor, the position of the content will be after the header because the header is not in a fixed or absolute position (you can try by yourself using https://tikui.org/atom/atom.html#button )